lib/cli-engine/config-array-factory.js

```diff
@@ -379,61 +401,91 @@ class ConfigArrayFactory {
-                ? this._normalizeConfigData(configData, filePath, name)
-                : null,
-            parent
-        );
+    create(configData, { basePath, filePath, name } = {}) {
+        if (!configData) {
+            return new ConfigArray();
+        }
+
+        const { cwd } = internalSlotsMap.get(this);
+        const ctx = createContext(cwd, "config", name, filePath, basePath);
+        const elements = this._normalizeConfigData(configData, ctx);
+
+        return new ConfigArray(...elements);
     }
 
     /**
      * Load a config file.
      * @param {string} filePath The path to a config file.
      * @param {Object} [options] The options.
+     * @param {string} [options.basePath] The base path to resolve relative paths in `overrides[].files`, `overrides[].excludedFiles`, and `ignorePatterns`.
      * @param {string} [options.name] The config name.
-     * @param {ConfigArray} [options.parent] The parent config array.
      * @returns {ConfigArray} Loaded config.
      */
-    loadFile(filePath, { name, parent } = {}) {
+    loadFile(filePath, { basePath, name } = {}) {
         const { cwd } = internalSlotsMap.get(this);
-        const absolutePath = path.resolve(cwd, filePath);
+        const ctx = createContext(cwd, "config", name, filePath, basePath);
 
-        return createConfigArray(
-            this._loadConfigData(absolutePath, name),
-            parent
-        );
+        return new ConfigArray(...this._loadConfigData(ctx));
     }
 
     /**
      * Load the config file on a given directory if exists.
      * @param {string} directoryPath The path to a directory.
      * @param {Object} [options] The options.
+     * @param {string} [options.basePath] The base path to resolve relative paths in `overrides[].files`, `overrides[].excludedFiles`, and `ignorePatterns`.
      * @param {string} [options.name] The config name.
-     * @param {ConfigArray} [options.parent] The parent config array.
      * @returns {ConfigArray} Loaded config. An empty `ConfigArray` if any config doesn't exist.
      */
-    loadInDirectory(directoryPath, { name, parent } = {}) {
+    loadInDirectory(directoryPath, { basePath, name } = {}) {
         const { cwd } = internalSlotsMap.get(this);
-        const absolutePath = path.resolve(cwd, directoryPath);
 
-        return createConfigArray(
-            this._loadConfigDataInDirectory(absolutePath, name),
-            parent
-        );
+        for (const filename of configFilenames) {
+            const ctx = createContext(
+                cwd,
+                "config",
+                name,
+                path.join(directoryPath, filename),
+                basePath
+            );
+
+            if (fs.existsSync(ctx.filePath)) {
+                let configData;
+
+                try {
+                    configData = loadConfigFile(ctx.filePath);
+                } catch (error) {
+                    if (!error || error.code !== "ESLINT_CONFIG_FIELD_NOT_FOUND") {
```

Comment:
Not a blocker for this PR, but an idea I had while reviewing this: maybe we could centralize these errors codes as constants that can be imported?

Reply:
I’d suggest opening an issue for that idea so we can focus the conversation on this PR around what needs to happen to get this merged.

Reply:
For information, this constant value should appear in the userland only if `--config path/to/a/package.json` is present, the `package.json` exists, and `eslintConfig` field doesn't exist in the `package.json`. (We can distinguish between `ENOENT` and that by code, but it may be not important.)